Lib/dis.py

```diff
@@ -32,25 +32,30 @@ def _try_compile(source, name):
     return c
 
 def dis(x=None, *, file=None, depth=None):
-    """Disassemble classes, methods, functions, generators,
-       asynchronous generators, coroutines, or code.
+    """Disassemble classes, methods, functions, other compiled objects, or code.
 
-    With no argument, disassemble the last traceback.
+       With no argument, disassemble the last traceback.
 
+       Compiled objects currently include generator objects, async generator
+       objects, and coroutine objects, all of which store their code object
+       in a special attribute.
```

Comment:
Reviewing this and thinking about how modules and classes are handled made me realise that there's now a discrepancy between the types that are accepted as arguments to `dis`, and those that will be disassembled when present as instance, class or module attributes.

Specifically, the filter for the `__dict__` handling case is defined in terms of `isinstance` and the `_have_code` type tuple, while `dis()` itself works entirely by checking for particular attributes.

Resolving that is out of scope for this PR, so I've filed a new issue to cover it: https://bugs.python.org/issue31197